packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -632,11 +640,14 @@ export class GarbageCollector implements IGarbageCollector {
         },
     ): Promise<IGCStats> {
         const {
-            logger = this.mc.logger,
             runSweep = this.shouldRunSweep,
             fullGC = this.gcOptions.runFullGC === true || this.summaryStateNeedsReset,
         } = options;
 
+        const logger = options.logger
```

Comment:
When is this set v. not?

Reply:
This is mostly for testing where you don't really need to pass a logger here.

Reply:
Might be simpler to always require it and have the tests do whatever they need to make it work?  Not for this change, just thinking out loud.